contracts/src/v0.8/feeds/DualAggregator.sol

```diff
@@ -716,41 +739,87 @@ contract DualAggregator is OCR2Abstract, OwnerIsCreator, AggregatorV2V3Interface
     bytes32[] calldata ss,
     bytes32 rawVs
   ) external override {
+    // Call the internal transmit function without the isSecondary flag
+    _transmit(
+      reportContext,
+      report,
+      rs,
+      ss,
+      rawVs,
+      false
+    );
+  }
+
+  // Secondary proxy transmit entrypoint, call the internal transmit function with the isSecondary flag
+  function transmitSecondary(
+    // reportContext consists of:
+    // reportContext[0]: ConfigDigest
+    // reportContext[1]: 27 byte padding, 4-byte epoch and 1-byte round
+    // reportContext[2]: ExtraHash
+    bytes32[3] calldata reportContext,
+    bytes calldata report,
+    // ECDSA signatures
+    bytes32[] calldata rs,
+    bytes32[] calldata ss,
+    bytes32 rawVs
+  ) external {
+    _transmit(
+      reportContext,
+      report,
+      rs,
+      ss,
+      rawVs,
+      true
+    );
+  }
+
+  // Internal transmit function with all the transmission logic
+  function _transmit(
+    // reportContext consists of:
+    // reportContext[0]: ConfigDigest
+    // reportContext[1]: 27 byte padding, 4-byte epoch and 1-byte round
+    // reportContext[2]: ExtraHash
+    bytes32[3] calldata reportContext,
+    bytes calldata report,
+    // ECDSA signatures
+    bytes32[] calldata rs,
+    bytes32[] calldata ss,
+    bytes32 rawVs,
+    bool isSecondary
+  ) internal {
     // NOTE: If the arguments to this function are changed, _requireExpectedMsgDataLength and/or
     // TRANSMIT_MSGDATA_CONSTANT_LENGTH_COMPONENT need to be changed accordingly
 
     uint256 initialGas = gasleft(); // This line must come first
-    Report memory report_;
 
-    if (msg.sender == s_secondaryProxy) {
-      // Decode the report
-      report_ = _decodeReport(report);
-      (bool exist, uint32 roundId) = _reportExists(report_);
-      // In case the report exists, only copy the round id
+    // Report epoch and round
+    uint40 epochAndRound = uint40(uint256(reportContext[1]));
+    // Validate the report data
+    _validateReport(epochAndRound, reportContext, report, rs, ss);
+
+    Report memory report_ = _decodeReport(report); // Decode the report
+
+    if (isSecondary) {
+      (bool exist, uint32 roundId) = _doesReportExist(report_);
+      // In case the report exists, copy the round id and pay the transmitter
       if (exist) {
         s_hotVars.latestSecondaryRoundId = roundId;
+        _payTransmitter(s_hotVars, report_.juelsPerFeeCoin, uint32(initialGas), msg.sender);
+
         return;
       } else {
```

Comment:
I think you can remove the else here since the if block returns. Makes it a bit more readable with one less nested block.